### PR TITLE
display reactions details

### LINKED ISSUE #2675: show reaction details
a tap in the reaction line (added in #2401) shall open reaction details (cmp. signal, whatsapp)

some implementation also allow to send reactions from there, however we do not want that but go for a unique way for sending reactions that is the same if there are no reactions at all (long tap, see #2402).

successor of #2401

### PR DESCRIPTION
**Pull request by adbenitez**
close #2675

### CODE REVIEW

**Comment by adbenitez** [comment]
preview:

image

**Comment by r10s** [comment]
it even updates when reaction change! 👌

File: res/values/strings.xml [hunk 1]
```diff
@@ -722,6 +722,9 @@
     <string name="emoji_symbols">Symbols</string>
     <string name="emoji_flags">Flags</string>
 
+    <!-- Reactions -->
+    <string name="reactions_details_title">Reactions</string>
```
Comment (r10s) [inline res/values/strings.xml:726]: just call it `reactions` - if we need to become more specific, we do that as needed

File: res/layout/reaction_recipient_item.xml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        android:id="@+id/reaction"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:gravity="center"
+        android:textSize="22dp"
+        android:fontFamily="sans-serif"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:text=":)" />
```
Comment (r10s) [inline res/layout/reaction_recipient_item.xml:79]: the padding right to the reaction should be the same as left of the avatar

to me, it looks too small

File: src/org/thoughtcrime/securesms/ConversationFragment.java [hunk 3]
```diff
@@ -867,6 +869,12 @@ public void onShowFullClicked(DcMsg messageRecord) {
       public void onDownloadClicked(DcMsg messageRecord) {
         dcContext.downloadFullMsg(messageRecord.getId());
       }
+
+      @Override
+      public void onReactionClicked(DcMsg messageRecord) {
+        ReactionsDetailsFragment dialog = new ReactionsDetailsFragment(messageRecord.getId());
+        dialog.show(((FragmentActivity) getActivity()).getSupportFragmentManager(), null);
```
Comment (r10s) [inline src/org/thoughtcrime/securesms/ConversationFragment.java:876]: the cast to `FragmentActivity` activity is not needed say android studio, and indeed it works without.

at line 50, `import androidx.fragment.app.FragmentActivity;` can be removed as well then

File: src/org/thoughtcrime/securesms/ConversationItem.java [hunk 4]
```diff
@@ -680,6 +680,10 @@ private void setReactions(@NonNull DcMsg current) {
         reactionsView.clear();
       } else {
         reactionsView.setReactions(reactions.getReactions());
+        reactionsView.setOnClickListener(v -> {
+          if (eventListener == null) return;
+          eventListener.onReactionClicked(current);
```
Comment (r10s) [inline src/org/thoughtcrime/securesms/ConversationItem.java:685]: the following style is used most often in this project ...

```suggestion
          if (eventListener != null) {
            eventListener.onReactionClicked(current);
          }
```

... and it is indeed easier to read in many cases as there are no different exit cases

File: src/org/thoughtcrime/securesms/reactions/ReactionRecipientsAdapter.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+import java.util.ArrayList;
+
+public class ReactionRecipientsAdapter extends RecyclerView.Adapter
+{
+  @SuppressWarnings("unused")
+  private final static String TAG = ReactionRecipientsAdapter.class.getSimpleName();
+
+  private @NonNull ArrayList<Pair<Integer, String>> contactsReactions = new ArrayList<>();
+  private final LayoutInflater                li;
```
Comment (r10s) [inline src/org/thoughtcrime/securesms/reactions/ReactionRecipientsAdapter.java:24]: while using short abbreviations if they're used only at a scope of few lines is okay (loops), for member variables, this is not-so-good style - eg. many will read this as "list item" at a first glance :)

better be verbose here and call the things what it is:


```suggestion
  private final LayoutInflater                layoutInflater;
```

(usages need to be adapted as well)

**Review by r10s — APPROVED** [review]
lgtm, good job! some minor comments only

**Comment by r10s** [comment]
another tiny thing: the reactions details should not open in multiselect-mode - instead, just select/deselect items (pass the tap to the parent or so)

**Comment by adbenitez** [comment]
> another tiny thing: the reactions details should not open in multiselect-mode - instead, just select/deselect items (pass the tap to the parent or so)

done :heavy_check_mark:

**Comment by r10s** [comment]
this pr needs a rebase on stable before being able to being merged

**Comment by adbenitez** [comment]
mmmh github shows no conflict :thinking:

**Comment by r10s** [comment]
> mmmh github shows no conflict 🤔

this is how github looks to me:

Screenshot 2023-09-30 at 16 06 30

**Comment by adbenitez** [comment]
LOL :dizzy_face:  this is how it looks to me:
image

**Comment by adbenitez** [comment]
well merging worked :thinking: maybe some github glitch on your side

**Comment by r10s** [comment]
anyway, it worked :)

**Pull request merged**